Integrations/FeedTAXII/FeedTAXII.py

```diff
@@ -503,16 +503,19 @@ def __init__(self, insecure: bool = True, polling_timeout: int = 20, initial_int
         self.collection = collection
 
         # authentication
-        if credentials:
-            if '_header:' in credentials.get('identifier', None):
-                self.api_header = credentials.get('identifier', None).split('_header:')[1]
-                self.api_key = credentials.get('password', None)
-            else:
-                self.username = credentials.get('identifier', None)
-                self.password = credentials.get('password', None)
+        if not credentials:
+            credentials = {}
+        self.api_head, self.api_key, self.username, self.password = '', '', '', ''
```

Comment:
i prefer we do it in the condition
so in the condition we will init `username` and `password` to be empty
in the `else` clause we will init `api_header` and `api_key`

Reply:
Unless I'm misunderstanding you - that still seems problematic - you could still have an `AttributeError` occur. Instance variables should be initialized in all cases, even if it's just to the empty string. I don't think we should be defining instance attributes on the fly. If I get an instance of a class, regardless of when or under what condition, there should be a well-defined list of instance attributes for that class, whether or not those instance attributes have non-null/0/empty-string values assigned to them or not, the _attributes_ themselves should always exist.